### PR TITLE
Futures history mobile display

### PR DESCRIPTION
**Pull request by Tburm**
Improve the display of the futures history table on mobile.

## Description
* Implements a specific mobile table for futures history
* Improves types and data handling of futures trade history
* Adds drawer to futures history on mobile
* Improves `PositionType` component to have a mobile version with no background

## Related issue
No issue for the original changes, but follow-up issues:
- #1479 
- #1480

## Screenshots (if appropriate):
image
image

### CODE REVIEW

File: components/Text/PositionType.tsx [hunk 1]
```diff
@@ -34,4 +41,22 @@ const StyledText = styled.p<{ $side: PositionProps['side'] }>`
 		`};
 `;
 
+const MobileStyledText = styled.p<{ side: PositionSide }>`
+	margin: 0;
+	text-transform: uppercase;
+	font-family: ${(props) => props.theme.fonts.bold};
+
+	${(props) =>
```
Comment (avclarke) [inline components/Text/PositionType.tsx:49]: I think these could be compressed to two lines with:
```
color: ${props => props.side === PositionSide.LONG ? props.theme.colors.common.primaryGreen : props.theme.colors.common. primaryRed}
```

File: sections/dashboard/FuturesHistoryTable/FuturesHistoryTable.tsx [hunk 2]
```diff
@@ -22,14 +23,23 @@ import useNetworkSwitcher from 'hooks/useNetworkSwitcher';
...
+	getDisplayAsset,
+	getMarketName,
+	isDecimalFour,
+	MarketKeyByAsset,
+} from 'utils/futures';
 
 import TimeDisplay from '../../futures/Trades/TimeDisplay';
 
 const FuturesHistoryTable: FC = () => {
+	const [selectedTrade, setSelectedTrade] = React.useState<FuturesTrade>();
```
Comment (avclarke) [inline sections/dashboard/FuturesHistoryTable/FuturesHistoryTable.tsx:42]: Be good to stick with avoiding `React.`

**Review by avclarke — COMMENTED** [review]
Looks good, just a couple of places for minor clean up.

File: components/Text/PositionType.tsx [hunk 3]
```diff
@@ -1,37 +1,57 @@
...
+			  `};
+	};
+`;
+
+const MobileStyledText = styled.p<{ side: PositionSide }>`
+	margin: 0;
+	text-transform: uppercase;
+	font-family: ${(props) => props.theme.fonts.bold};
 
 	${(props) =>
```
Comment (avclarke) [inline components/Text/PositionType.tsx:47]: Also a bit boated here, could reduce to:
```
color: ${(props) => props.side === PositionSide.LONG ? props.theme.colors.common.primaryGreen : props.theme.colors.common.primaryRed}
```

**Review by avclarke — DISMISSED** [review]
Not a blocker but just noticed where the conditional styling is still bloated.

File: components/Text/PositionType.tsx (hunk 3 above)
Comment (Tburm) [reply components/Text/PositionType.tsx]: fixed this and one other one I hadn't noticed where we repeated the styles for both sides.

**Comment by Tburm** [comment]
@asherism I addressed your comments, but I left the "spot history" table alone right now. I wrote a separate issue to fix that table, since we probably need a specific mobile design. It's doing the same thing running over the side of the container.

**Comment by asherism** [comment]
> @asherism I addressed your comments, but I left the "spot history" table alone right now. I wrote a separate issue to fix that table, since we probably need a specific mobile design. It's doing the same thing running over the side of the container.

sounds good. another thing i wanted to mention- we should truncate the 24h vol numbers after 100k or 1m, open interest as well

**Comment by platschi** [comment]
> > @asherism I addressed your comments, but I left the "spot history" table alone right now. I wrote a separate issue to fix that table, since we probably need a specific mobile design. It's doing the same thing running over the side of the container.
> 
> sounds good. another thing i wanted to mention- we should truncate the 24h vol numbers after 100k or 1m, open interest as well

Noted, will ticket that out, sounds like a quick devdao ticket. Will test & merge this PR otherwise.

**Review by platschi — APPROVED** [review]

**Pull request merged**